### PR TITLE
Unit test for index and show of book

### PR DESCRIPTION
**Pull request by Gmarn**
# Summary

- Adding an factory for user
- Updating the rails_helper
- Editing test for the book controller

# Trello card

https://trello.com/c/ja8CJ3PS

### CODE REVIEW

**Comment by alanhala** [comment]
@Gmarn Please, rebase this branch. If you have any troubles, tell me and I help you

File: app/controllers/api/v1/rents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Api
```
Comment (Dinuuu) [inline app/controllers/api/v1/rents_controller.rb]: why did you add this to this commit that is about testing ?

Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: I do not know, I need it to one an other card next but I can not remember why this is here...

Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: I can delete it for the moment

Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: It seem I do not need it, in fact, after. So I delete it and i change the route.rb file too.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -63,3 +64,14 @@
     with.library :rails
   end
 end
+
+RSpec.shared_context 'Authenticated User' do
```
Comment (francoprud) [inline spec/rails_helper.rb]: Place this in a new file inside `spec/support/`

Comment (francoprud) [inline spec/rails_helper.rb]: Also add the following to this file under the requires

```ruby
Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
```

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@
 abort("The Rails environment is running in production mode!") if Rails.env.production?
 require 'spec_helper'
 require 'rspec/rails'
+require 'support/factory_girl'
```
Comment (francoprud) [inline spec/rails_helper.rb]: Remove this include and that file.

Add that line inside the `RSpec.config` from the `spec_helper.rb`

File: spec/controllers/api/v1/books_controller.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+describe Api::V1::BooksController, type: :controller do
```
Comment (francoprud) [inline spec/controllers/api/v1/books_controller.rb]: `:type` is not necessary

File: app/controllers/api/v1/books_controller.rb [hunk 5]
```diff
@@ -2,13 +2,13 @@ module Api
   module V1
     class BooksController < ApiController
       def index
-        @books = Book.all
-        render json: @books, each_serializer: IndexBookSerializer
+        books = Book.all
+        render json: books, each_serializer: IndexBookSerializer
```
Comment (francoprud) [inline app/controllers/api/v1/books_controller.rb]: Why this 2 different serializers?

Comment (francoprud) [inline app/controllers/api/v1/books_controller.rb]: Why not BookSerializer? Also the use of serializers is inferred
